plugin/evm/header/base_fee_test.go

```diff
@@ -11,328 +11,344 @@ import (
+			require := require.New(t)
 
-	header := &types.Header{
-		Time:    0,
-		GasUsed: 1_000_000,
-		Number:  big.NewInt(0),
-		BaseFee: big.NewInt(225 * params.GWei),
-		Extra:   nil,
-	}
-	extDataHeader := &types.Header{
-		Time:    0,
-		GasUsed: 1_000_000,
-		Number:  big.NewInt(0),
-		BaseFee: big.NewInt(225 * params.GWei),
-		Extra:   nil,
-		// ExtDataGasUsage is set to be nil to ensure CalcBaseFee can handle the
-		// AP3/AP4 boundary.
-	}
-
-	for index, event := range events {
-		block := event.block
-		nextExtraData, err := ExtraPrefix(params.TestApricotPhase4Config, header, block.timestamp)
-		assert.NoError(t, err)
-		nextBaseFee, err := BaseFee(params.TestApricotPhase4Config, header, block.timestamp)
-		assert.NoError(t, err)
-		log.Info("Update", "baseFee", nextBaseFee)
-		header = &types.Header{
-			Time:    block.timestamp,
-			GasUsed: block.gasUsed,
-			Number:  big.NewInt(int64(index) + 1),
-			BaseFee: nextBaseFee,
-			Extra:   nextExtraData,
-		}
-
-		nextExtraData, err = ExtraPrefix(params.TestApricotPhase4Config, extDataHeader, block.timestamp)
-		assert.NoError(t, err)
-		nextBaseFee, err = BaseFee(params.TestApricotPhase4Config, extDataHeader, block.timestamp)
-		assert.NoError(t, err)
-		log.Info("Update", "baseFee (w/extData)", nextBaseFee)
-		extDataHeader = &types.Header{
-			Time:           block.timestamp,
-			GasUsed:        block.gasUsed,
-			Number:         big.NewInt(int64(index) + 1),
-			BaseFee:        nextBaseFee,
-			Extra:          nextExtraData,
-			ExtDataGasUsed: block.extDataGasUsed,
-		}
-
-		assert.Equal(t, event.extDataFeeGreater, extDataHeader.BaseFee.Cmp(header.BaseFee) == 1, "unexpected cmp for index %d", index)
-	}
-}
-
-func TestDynamicFeesEtna(t *testing.T) {
-	require := require.New(t)
-	header := &types.Header{
-		Number: big.NewInt(0),
-	}
-
-	timestamp := uint64(1)
-	extra, err := ExtraPrefix(params.TestEtnaChainConfig, header, timestamp)
-	require.NoError(err)
-	nextBaseFee, err := BaseFee(params.TestEtnaChainConfig, header, timestamp)
-	require.NoError(err)
-	// Genesis matches the initial base fee
-	require.Equal(int64(ap3.InitialBaseFee), nextBaseFee.Int64())
-
-	timestamp = uint64(10_000)
-	header = &types.Header{
-		Number:  big.NewInt(1),
-		Time:    header.Time,
-		BaseFee: nextBaseFee,
-		Extra:   extra,
-	}
-	nextBaseFee, err = BaseFee(params.TestEtnaChainConfig, header, timestamp)
-	require.NoError(err)
-	// After some time has passed in the Etna phase, the base fee should drop
-	// lower than the prior base fee minimum.
-	require.Less(nextBaseFee.Int64(), int64(ap4.MinBaseFee))
-}
-
-func TestCalcBaseFeeRegression(t *testing.T) {
-	parentTimestamp := uint64(1)
-	timestamp := parentTimestamp + ap3.WindowLen + 1000
+			config := &params.ChainConfig{
+				NetworkUpgrades: test.upgrades,
+			}
+			got, err := BaseFee(config, test.parent, test.timestamp)
+			require.ErrorIs(err, test.wantErr)
+			require.Equal(test.want, got)
 
-	parentHeader := &types.Header{
```

Comment:
should we keep this regression test? (can be a case in the new test)
I see we kept the assertion but not sure that we have the conditions that triggered this.

Reply:
This test was migrated to the new test as `ap3_big_1_not_modified`